res/controllers/Denon-MC7000-scripts.js

```diff
@@ -833,10 +993,37 @@ MC7000.crossfaderAssign = function(channel, control, value, status, group) {
 // Assign Spinback length to STOP TIME knob
 MC7000.stopTime = function(channel, control, value, status, group) {
     var deckNumber = script.deckFromGroup(group);
-    // "factor" for engine.brake()
+    // "factor" for engine.brake() and engine.softStart()
     // this formula produces factors between 31 (min STOP TIME for ca 7 sec back
     // in track) and 1 (max STOP TIME for ca 18.0 sec back in track)
     MC7000.factor[deckNumber] = (1.1 - (value / 127)) * 30 - 2;
+    MC7000.factor2[deckNumber] = (127.69 - value);
+};
+MC7000.lastpress = 0;
+MC7000.play = function(channel, control, value, status, group) {
+    if (value === 0x00) {
+        return; // don't respond to note off messages
+    }
+    var deckNumber = script.deckFromGroup(group);
+    // set a variable to toggle between play and pause, based on current play status
+    var playToggle = 1 - engine.getValue(group, "play_indicator");
```

Comment:
[`play_indicator`](https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-play_indicator) is only for direct UI output (it changes behavior based on Cue mode). You'll probably want `play` instead. Also why the `1 -`?
```suggestion
    // set a variable to toggle between play and pause, based on current play status
    var playToggle = engine.getValue(group, "play");
```

Reply:
changed